libs/feature/search/src/lib/metadata-view/links/link-helper.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core'
+import { DatasetFinderService, LinkUsage } from '@geonetwork-ui/feature/dataviz'
+import { MetadataLink, MetadataRecord } from '@geonetwork-ui/util/shared'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LinkHelperService {
+  constructor(private datasetFinder: DatasetFinderService) {}
+
+  hasLinks(record: MetadataRecord): boolean {
+    return 'links' in record
+  }
+  isDataLink(link: MetadataLink): boolean {
+    return this.datasetFinder.getLinkUsages(link).length > 0
+  }
+  isOtherLink(link: MetadataLink): boolean {
+    return this.datasetFinder.getLinkUsages(link).length === 0
+  }
+  isValidLink(link: MetadataLink): boolean {
+    return !('invalid' in link)
+  }
+  isMapLink(link: MetadataLink): boolean {
+    return this.datasetFinder.getLinkUsages(link).indexOf(LinkUsage.MAP) > -1
+  }
+  isDownloadLink(link: MetadataLink): boolean {
+    return (
+      this.datasetFinder.getLinkUsages(link).indexOf(LinkUsage.DOWNLOAD) > -1
+    )
+  }
+}
```

Comment:
To me this could almost be simple function exports without an injectable service. Would make usage less verbose! Just a suggestion.

Reply:
I wanted to to this but as I need the `datasetFinder` which is a service, so i need to create a service to inject the `datasetFinder`

Reply:
Ah, true, too bad then :)